ibctest/ibc_test.go

```diff
@@ -1,296 +1,62 @@
-	t *testing.T,
-	pool *dockertest.Pool,
-	networkID string,
-	home string,
-) ibc.Relayer {
-	r := &relayer{
-		t: t,
-
-		pool: pool,
-
-		home: home,
-	}
-
-	res := r.sys().Run(zaptest.NewLogger(t), "config", "init")
-	if res.Err != nil {
-		panic(fmt.Errorf("failed to rly config init: %w", res.Err))
-	}
-
-	return r
-}
-
-func (relayerFactory) Capabilities() map[ibctestrelayer.Capability]bool {
-	// It is currently expected that the main branch of the relayer supports all tested features.
-	return ibctestrelayer.FullCapabilities()
-}
-
-func (relayerFactory) Labels() []label.Relayer {
-	return []label.Relayer{label.Rly}
-}
-
-func (relayerFactory) Name() string { return "github.com/cosmos/relayer" }
-
-type relayer struct {
-	t *testing.T
-
-	pool *dockertest.Pool
-
-	home string
-
-	// Set during StartRelayer.
-	errCh  chan error
-	cancel context.CancelFunc
-}
-
-func (r *relayer) sys() *relayertest.System {
-	return &relayertest.System{HomeDir: r.home}
-}
-
-func (r *relayer) log() *zap.Logger {
-	return zaptest.NewLogger(r.t)
-}
-
-func (r *relayer) AddChainConfiguration(ctx context.Context, _ ibc.RelayerExecReporter, chainConfig ibc.ChainConfig, keyName, rpcAddr, grpcAddr string) error {
-	sys := &relayertest.System{HomeDir: r.home}
-	sys.MustAddChain(r.t, chainConfig.ChainID, cmd.ProviderConfigWrapper{
-		Type: "cosmos",
-		Value: cosmos.CosmosProviderConfig{
-			Key:     keyName,
-			ChainID: chainConfig.ChainID,
-			RPCAddr: rpcAddr,
-			// GRPCAddr: grpcAddr, // Not part of relayer cosmos provider config (yet)
-			AccountPrefix:  chainConfig.Bech32Prefix,
-			KeyringBackend: keyring.BackendTest,
-			GasAdjustment:  chainConfig.GasAdjustment,
-			GasPrices:      chainConfig.GasPrices,
-			Debug:          true,
-			Timeout:        "10s",
-			OutputFormat:   "json",
-			SignModeStr:    "direct",
-		},
-	})
-
-	return nil
-}
-
-func (r *relayer) AddKey(ctx context.Context, _ ibc.RelayerExecReporter, chainID, keyName string) (ibc.RelayerWallet, error) {
-	res := r.sys().RunC(ctx, r.log(), "keys", "add", chainID, keyName)
-	if res.Err != nil {
-		return ibc.RelayerWallet{}, res.Err
-	}
-
-	var w ibc.RelayerWallet
-	if err := json.Unmarshal(res.Stdout.Bytes(), &w); err != nil {
-		return ibc.RelayerWallet{}, err
-	}
-
-	return w, nil
-}
-
-func (r *relayer) RestoreKey(ctx context.Context, _ ibc.RelayerExecReporter, chainID, keyName, mnemonic string) error {
-	res := r.sys().RunC(ctx, r.log(), "keys", "restore", chainID, keyName, mnemonic)
-	if res.Err != nil {
-		return res.Err
-	}
-	return nil
-}
-
-func (r *relayer) GeneratePath(ctx context.Context, _ ibc.RelayerExecReporter, srcChainID, dstChainID, pathName string) error {
-	res := r.sys().RunC(ctx, r.log(), "paths", "new", srcChainID, dstChainID, pathName)
-	if res.Err != nil {
-		return res.Err
-	}
-	return nil
+// TestRelayerBinary runs the ibctest conformance tests against
+// the current state of this relayer implementation built as a binary.
+func TestRelayerBinary(t *testing.T) {
```

Comment:
This is inaccurate -- there is no standalone binary built in this approach. `TestRelayerInProcess` would be more precise.

Reply:
fixed